go/obscuronode/enclave/enclave.go

```diff
@@ -575,6 +575,34 @@ func (e *enclaveImpl) GetTransaction(txHash common.Hash) *nodecommon.L2Tx {
 	}
 }
 
+func (e *enclaveImpl) GetTransactionReceipt(encryptedParams nodecommon.EncryptedParamsGetTxReceipt) (nodecommon.EncryptedResponseGetTxReceipt, error) {
+	txHash, err := e.extractTxHash(encryptedParams)
+	if err != nil {
+		return nil, err
+	}
+
+	viewingKeyAddress, err := e.getTxSender(txHash)
+	if err != nil {
+		return nil, err
+	}
+
+	txReceipt, err := e.storage.GetTransactionReceipt(txHash)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve transaction receipt in eth_getTransactionReceipt request. Cause: %w", err)
+	}
+	txReceiptBytes, err := txReceipt.MarshalJSON()
```

Comment:
feels like marshaling belongs logically with the encryption.

Reply:
@tudor-malene If I do that, I need a separate encryption function for each type (balance, call, tx receipt).